### PR TITLE
feat: Logging in with phone

### PR DESCRIPTION
**Pull request by gorandalum**

### CODE REVIEW

File: ios/atb/Info.plist [hunk 1]
```diff
@@ -32,6 +32,14 @@
...
 			</array>
 		</dict>
+		<dict>
+			<key>CFBundleTypeRole</key>
+			<string>Editor</string>
+			<key>CFBundleURLSchemes</key>
+			<array>
+				<string>com.googleusercontent.apps.827196677776-fft65c0v4jp8u7fdlqo03ep1u0n66fhf</string>
+			</array>
+		</dict>
```
Comment (cbrevik) [inline ios/atb/Info.plist:42]: Where does this come in?

Comment (gorandalum) [reply ios/atb/Info.plist:42]: It's for setting up recaptcha verification on iOS. Described here:
https://firebase.google.com/docs/auth/ios/phone-auth#set-up-recaptcha-verification

Comment (cbrevik) [reply ios/atb/Info.plist:42]: Aha. That value seems to be unique per bundle identifier (`debug`, `staging`, `store`), so we'll probably need to override it per environment then too: https://github.com/AtB-AS/mittatb-app/blob/8b249184bdf8c14ec15b1a4b92559dba77e83af7/scripts/override-config-files.sh#L16

Comment (cbrevik) [reply ios/atb/Info.plist:42]: On iOS, we have a unique `GoogleService-Info.plist` per env (see `ios`-folder, as well as each folder under `env`). So we can probably extract `REVERSED_CLIENT_ID` from those, then set it in our `Info.plist` as with the other API-keys, using `PlistBuddy`

**Comment by cbrevik** [comment]
image

Got this first time logging in on Simulator, do you notice it too? Seems to be user access issues when accessing Firestore-snapshot of tickets. Haven't found out what causes it yet, but probably a race condition between users (anonymous and phone user), and which has access to what?

File: src/screens/Profile/Login/ConfirmCode.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  return (
+    <View style={styles.container}>
+      <FullScreenHeader title={'Logg inn'} leftButton={{type: 'back'}} />
+
+      <View style={styles.mainView}>
+        <Sections.Section>
+          <Sections.TextInput
+            label={'Kode'}
+            placeholder="Bekreftelseskode fra SMS"
+            onChangeText={setCode}
```
Comment (cbrevik) [inline src/screens/Profile/Login/ConfirmCode.tsx:35]: Should also use phone pad here, as well as hint to the OS that this input expects an OTP

```suggestion
            onChangeText={setCode}
            keyboardType="phone-pad"
            textContentType="oneTimeCode"
```

**Review by cbrevik — CHANGES_REQUESTED** [review]
A couple of suggested changes + something we should discuss

File: src/screens/Profile/Login/PhoneInput.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+};
+
+export default function PhoneInput({navigation}: PhoneInputProps) {
+  const {t} = useTranslation();
+  const styles = useThemeStyles();
+  const {signInWithPhoneNumber} = useAuthState();
+  const [phoneNumber, setPhoneNumber] = useState('');
+
+  const onNext = async () => {
+    await signInWithPhoneNumber(phoneNumber);
```
Comment (cbrevik) [inline src/screens/Profile/Login/PhoneInput.tsx:25]: Might want to trigger a loading event/indicator before this. It takes a bit of time to navigate to the next screen.

**Comment by cbrevik** [comment]
Testing on Android and works great with the test number! 🌮 🎉 

I don't have a SIM for my test-device, so unable to test a real OTP. But I'm sure one of the testers in QA when we get there can verify if it works.

File: src/screens/Profile/Login/ConfirmCode.tsx (hunk 2 above)
Comment (gorandalum) [reply src/screens/Profile/Login/ConfirmCode.tsx:80]: Fixed

File: src/screens/Profile/Login/PhoneInput.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    <View style={styles.container}>
+      <FullScreenHeader title={'Logg Inn'} leftButton={{type: 'cancel'}} />
+
+      <View style={styles.mainView}>
+        <Sections.Section>
+          <Sections.GenericItem>
+            <ThemeText>Telefonnummer</ThemeText>
+          </Sections.GenericItem>
+          <Sections.TextInput
+            label={'+ 47'}
```
[the remark this answers is not in the document]
Comment (gorandalum) [reply src/screens/Profile/Login/PhoneInput.tsx:56]: Petter suggested using a dropdown with country codes, but hardcoding it to +47 is good enough at this time.

File: src/screens/Profile/Login/PhoneInput.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            value={phoneNumber}
+            onChangeText={setPhoneNumber}
+            showClear={true}
```
[the remark this answers is not in the document]
Comment (gorandalum) [reply src/screens/Profile/Login/PhoneInput.tsx:59]: Fixed

File: ios/atb/Info.plist (hunk 1 above)
Comment (gorandalum) [reply ios/atb/Info.plist]: Fixed

File: src/screens/Profile/Login/PhoneInput.tsx (hunk 3 above)
Comment (gorandalum) [reply src/screens/Profile/Login/PhoneInput.tsx]: Fixed

**Comment by gorandalum** [comment]
> Got this first time logging in on Simulator, do you notice it too? Seems to be user access issues when accessing Firestore-snapshot of tickets. Haven't found out what causes it yet, but probably a race condition between users (anonymous and phone user), and which has access to what?

Yeah. I saw this too. Seems to me as a user access issue right at the time were the user changes from logged in as anonymous too logged in with phone or vica versa. What should be done? Resetting the `abtCustomerId` when changing login method?

**Review by cbrevik — APPROVED** [review]
As noted on Slack, translations + `Info.plist` should be done before merging.  Other than that it LGTM 💯. Good job!

File: src/screens/Profile/Login/PhoneInput.tsx (hunk 3 above)
Comment (mikaelbr) [reply src/screens/Profile/Login/PhoneInput.tsx]: Maybe check out `useIsLoading` (in `@atb/utils/use-is-loading.ts`)

File: src/translations/screens/subscreens/Login.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
+import {translation as _} from '../../commons';
+const LoginTexts = {
```
Comment (gorandalum) [inline src/translations/screens/subscreens/Login.ts:2]: @magnusolderoesaether @pettersuul 
You can take a look at the texts and the translations in this file.

**Pull request merged**